Tests/SentryTests/Integrations/Performance/UIViewController/SentryTimeToDisplayTrackerTest.swift

```diff
@@ -0,0 +1,218 @@
+        let sut = fixture.getSut(for: UIViewController(), waitForFullDisplay: true)
+        let tracer = fixture.tracer
+
+        sut.start(for: tracer)
+
+        fixture.dateProvider.setDate(date: Date(timeIntervalSince1970: 10))
+        sut.reportReadyToDisplay()
+        fixture.displayLinkWrapper.normalFrame()
+
+        fixture.dateProvider.setDate(date: Date(timeIntervalSince1970: 11))
+        sut.reportFullyDisplayed()
+
+        XCTAssertNotNil(sut.fullDisplaySpan)
+        XCTAssertEqual(sut.fullDisplaySpan?.startTimestamp, Date(timeIntervalSince1970: 9))
+        XCTAssertEqual(sut.fullDisplaySpan?.timestamp, fixture.dateProvider.date())
+        XCTAssertEqual(sut.fullDisplaySpan?.status, .ok)
+
+        XCTAssertEqual(sut.fullDisplaySpan?.spanDescription, "UIViewController full display")
+        XCTAssertEqual(sut.fullDisplaySpan?.operation, SentrySpanOperationUILoadFullDisplay)
+    }
+
+    func testReportFullDisplay_waitingForFullDisplay_notReadyToDisplay() {
+        let sut = fixture.getSut(for: UIViewController(), waitForFullDisplay: true)
+        let tracer = fixture.tracer
+
+        sut.start(for: tracer)
+
+        fixture.displayLinkWrapper.normalFrame()
+
+        fixture.dateProvider.setDate(date: Date(timeIntervalSince1970: 11))
+        sut.reportFullyDisplayed()
+
+        XCTAssertFalse(sut.fullDisplaySpan?.isFinished ?? true)
+    }
+
+    func testReportFullDisplay_expires() {
+        fixture.dateProvider.setDate(date: Date(timeIntervalSince1970: 9))
+
+        let sut = fixture.getSut(for: UIViewController(), waitForFullDisplay: true)
+        let tracer = fixture.tracer
+
+        sut.start(for: tracer)
+
+        fixture.dateProvider.setDate(date: Date(timeIntervalSince1970: 10))
+        sut.reportReadyToDisplay()
+        fixture.displayLinkWrapper.normalFrame()
+
+        fixture.dateProvider.setDate(date: Date(timeIntervalSince1970: 11))
+        sut.fullDisplaySpan?.finish(status: .deadlineExceeded)
+
+        fixture.dateProvider.setDate(date: Date(timeIntervalSince1970: 13))
+        tracer.finish()
+
+        XCTAssertEqual(sut.fullDisplaySpan?.startTimestamp, Date(timeIntervalSince1970: 9))
+        XCTAssertEqual(sut.fullDisplaySpan?.timestamp, Date(timeIntervalSince1970: 10))
+        XCTAssertEqual(sut.fullDisplaySpan?.spanDescription, "UIViewController full display - Expired")
+    }
+
+    func test_checkInitialTime() {
+        fixture.dateProvider.setDate(date: Date(timeIntervalSince1970: 9))
+        fixture.dateProvider.driftTimeForEveryRead = true
+
+        let sut = fixture.getSut(for: UIViewController(), waitForFullDisplay: true)
+        let tracer = fixture.tracer
+
+        sut.start(for: tracer)
+
+        XCTAssertNotNil(sut.fullDisplaySpan)
+        XCTAssertEqual(sut.fullDisplaySpan?.startTimestamp, tracer.startTimestamp)
+        XCTAssertEqual(sut.initialDisplaySpan?.startTimestamp, tracer.startTimestamp)
+    }
+
+    func test_fullDisplay_reportedBefore_initialDisplay() {
```

Comment:
`l`: It would be great to stick to one naming pattern in the same test file.
```suggestion
    func testFullDisplay_reportedBefore_initialDisplay() {
```